File: react-admin-dashboard-master/src/services/InventoryService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8082/api/inventory';

// Récupérer tous les inventaires
export const getInventories = () => axios.get(REST_API_BASE_URL + '/all');

// Ajouter un inventaire
export const addInventory = (inventory) => {
    return axios.post(REST_API_BASE_URL, inventory);
};

// Récupérer un inventaire par ID
export const getInventory = (inventoryId) => axios.get(`${REST_API_BASE_URL}/${inventoryId}`);

// Mettre à jour un inventaire
export const updateInventory = (id, inventory) => {
    return axios.put(`${REST_API_BASE_URL}/${id}`, inventory);
};

// Supprimer un inventaire
export const deleteInventory = (id) => {
    return axios.delete(`${REST_API_BASE_URL}/${id}`);
};

 

File: spring-boot-microservices-part-2/inventory-service/src/main/java/com/programmingtechie/inventoryservice/service/InventoryService.java
package com.programmingtechie.inventoryservice.service;

import com.programmingtechie.inventoryservice.dto.InventoryResponse;
import com.programmingtechie.inventoryservice.model.Inventory;
import com.programmingtechie.inventoryservice.repository.InventoryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;
import org.springframework.http.HttpStatus;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class InventoryService {

    private final InventoryRepository inventoryRepository;

    @Transactional(readOnly = true)
    public List<InventoryResponse> isInStock(List<String> skuCode) {
        return inventoryRepository.findBySkuCodeIn(skuCode).stream()
                .map(inventory ->
                        InventoryResponse.builder()
                                .skuCode(inventory.getSkuCode())
                                .isInStock(inventory.getQuantity() > 0)
                                .quantity(inventory.getQuantity())
                          
                                .build()
                ).toList();
    }

    // Ajouter un inventaire
    public Inventory addInventory(Inventory inventory) {
        return inventoryRepository.save(inventory);
    }

    // Mettre à jour un inventaire
    public Inventory updateInventory(Long id, Inventory inventory) {
        if (inventoryRepository.existsById(id)) {
            inventory.setId(id);
            return inventoryRepository.save(inventory);
        }
        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Inventory with ID " + id + " not found");
    }

    // Supprimer un inventaire
    public boolean deleteInventory(Long id) {
        if (inventoryRepository.existsById(id)) {
            inventoryRepository.deleteById(id);
            return true;
        }
        return false;
    }

    // Récupérer tous les inventaires
    public List<Inventory> getAllInventories() {
        return inventoryRepository.findAll();
    }

    // Récupérer un inventaire spécifique par son ID
    public Optional<Inventory> getInventoryById(Long id) {
        return inventoryRepository.findById(id);
    }
}
